src/session/content/chat_action_bar.rs

```diff
@@ -282,14 +286,167 @@ impl ChatActionBar {
         self.save_message_as_draft();
 
         let imp = self.imp();
+        let mut bindings = imp.bindings.borrow_mut();
+        while let Some(binding) = bindings.pop() {
+            binding.unwatch();
+        }
 
         if let Some(ref chat) = chat {
             self.load_draft_message(chat.draft_message());
 
             imp.chat_action_in_cooldown.set(false);
+
+            let chat_expression = gtk::ConstantExpression::new(chat).upcast();
```

Comment:
Is this still required? since this variable seems to be unused
```rust
let permissions_expression = Chat::this_expression("permissions");
let blocked_expression = Chat::this_expression("is-blocked");
```